Orange/widgets/data/owcolor.py

```diff
@@ -350,6 +350,7 @@ def set_data(self, data):
                                         self._create_proxies(domain.metas))
             self.openContext(data)
             self.data = data.transform(domain)
+            self.data.domain = domain
```

Comment:
I guess we need a comment here, at the very least.

But there may be a more general problem here. One would expect that after `self.data = data.transform(domain)`, we would have `self.data.domain is domain`, while `transform` only ensures `self.data.domain == domain`. We should either change `transform` to (always) compare domains with `is`, not `==`, or we can add a flag to specify how to compare domains.

@lanzagar, @astaric?